src/hb-ot-color-cbdt-table.hh

```diff
@@ -243,6 +435,101 @@ struct IndexSubtableArray
+    if (unlikely (!dst)) return_trace (false);
+
+    hb_vector_t<hb_pair_t<hb_codepoint_t, const IndexSubtableRecord*>> lookup;
+    for (hb_codepoint_t new_gid = 0; new_gid < c->plan->num_output_glyphs (); new_gid++)
+    {
+      hb_codepoint_t old_gid;
+      if (unlikely (!c->plan->old_gid_for_new_gid (new_gid, &old_gid))) continue;
+
+      const IndexSubtableRecord* record = find_table(old_gid, original_num_tables);
+      if (unlikely (!record)) continue;
+      lookup.push (hb_pair_t<hb_codepoint_t, const IndexSubtableRecord*> (new_gid, record));
+
+      if (*start_glyph > *end_glyph)
+        *start_glyph = new_gid;
+
+      *end_glyph = new_gid;
+    }
+    if (unlikely (!lookup.length)) return_trace (false);
+
+    IndexSubtableRecord *curr_record = nullptr;
+    IndexSubtable *curr_subtable = nullptr;
+    bool success = true;
+    unsigned int num_glyphs = 0;
+    *size = 0;
+    *num_tables = 1;
+    if (unlikely (!lookup[0].second->create_new_record (c->serializer,
+                                                        cbdt->length,
+                                                        this,
+                                                        size,
+                                                        &curr_record,
+                                                        &curr_subtable)))
+      return_trace (false);
+    if (unlikely (!curr_record || !curr_subtable)) return_trace (false);
+
+    for (unsigned int i = 0; i < lookup.length; i++)
+    {
+      // On a match in formats we can extend the old table.
+      const IndexSubtable *old_subtable = lookup[i].second->get_subtable (this);
+      auto* new_header = curr_subtable->get_header ();
+      auto* old_header = old_subtable->get_header ();
+      if (!success ||
+          new_header->indexFormat != old_header->indexFormat ||
+          new_header->imageFormat != old_header->imageFormat)
+      {
+        curr_subtable->finish_subtable (c->serializer, cbdt->length, num_glyphs, size);
+        curr_record->finish_record (c->serializer, success, dst);
+
+        if (unlikely (!success)) return_trace (false);
+        if (unlikely (!lookup[i].second->create_new_record (c->serializer,
+                                                            cbdt->length,
+                                                            this,
+                                                            size,
+                                                            &curr_record,
+                                                            &curr_subtable)))
+          return_trace (false);
+        if (unlikely (!curr_record || !curr_subtable)) return_trace (false);
+
+        *num_tables += 1;
+        num_glyphs = 0;
+      }
+      hb_codepoint_t new_gid = lookup[i].first;
+      unsigned int num_missing = curr_record->add_glyph_for_subset (new_gid);
+      if (unlikely (!curr_subtable->fill_missing_glyphs (c->serializer, cbdt->length, num_missing, size, &num_glyphs)))
```

Comment:
If you implement the comment above then there shouldn't be any gaps and this becomes unnecessary.

Reply:
> I haven't seen a case where gaps occur so maybe we can just drop this though?

That's not a good way to write defensive code. :)